frontend/src/game/PhaserGame.tsx

```diff
@@ -109,22 +108,46 @@ export const PhaserGame = forwardRef<IRefPhaserGame, IProps>(function PhaserGame
   if (calculateVariableValue(gameData.ups.acquired, GameVariable.ButtonAvailable) === 1) {
     visible = true;
   }
-
-  return (
-    <div style={{ flex: 1, textAlign: "center", width: "100%", height: "100%" }}>
-      <button
-        style={{
-          visibility: visible ? "visible" : "hidden",
-          marginBottom: "1%",
-          width: "15%",
-          height: "5%",
-          fontSize: "18px"
-        }}
-        onClick={() => EventBus.emit("collect-all")}
-      >
-        Collect All
-      </button>
-      <div id="game-container" style={{ flex: 1, textAlign: "center", width: "100%", height: "90%" }}></div>
-    </div>
-  );
+  function reduceCooldown() {
+    setCooldown(cooldown - 1);
+    dispatch(setPreviousCooldown(cooldown));
+  }
+  if (cooldown === 0) {
```

Comment:
Both of these returns are almost identical, I'd recommend doing something like this instead,

const reduceCooldown = (): void => {
    setCooldown(cooldown - 1);
    dispatch(setPreviousCooldown(cooldown));
  };

  let innerTextNode: string;
  let disabledProp: boolean;
  let onClickProp: React.MouseEventHandler<HTMLButtonElement>;

  if (cooldown === 0) {
    onClickProp = () => EventBus.emit("collect-all") && setCooldown(5);
    innerTextNode = "Collect All";
  } else {
    disabledProp = true;
    innerTextNode = `Cooldown ${cooldown}`;
    setTimeout(reduceCooldown, 1000);
  }

  return (
    <div style={{ flex: 1, textAlign: "center", width: "100%", height: "100%" }}>
      <button
        style={{
          visibility: visible ? "visible" : "hidden",
          marginBottom: "1%",
          width: "15%",
          height: "5%",
          fontSize: "18px"
        }}
        onClick={onClickProp}
        disabled={disabledProp}
      >
        {innerTextNode}
      </button>
      <div id="game-container" style={{ flex: 1, textAlign: "center", width: "100%", height: "90%" }}></div>
    </div>
  );


![image](https://github.com/user-attachments/assets/49edf7e8-7045-417f-98f6-3f4b4cd9a25a)